File: CppProgramm/Computer.cpp
#include "Computer.h"

void Computer::init_computer(string name, Processor processor, Ram ram, Videocard videocard, Motherboard motherboard) {  //создание компьютера
	this->name = name;
	this->processor = processor;
	this->ram = ram;
	this->videocard = videocard;
	this->motherboard = motherboard;
}

void Computer::turn_on() {  //включить компьютер
	if (state == 0) { state = 1; cout << "Компьютер включен: " << name << endl; }
	else cout << "Компьютер уже включен: " << name << endl;
}

void Computer::turn_off() {  //выключить компьютер
	if (state == 1) { state = 0; cout << "Компьютер выключен: " << name << endl; }
}

void Computer::check() {  //проверить и вывести состовляющие компьютера на экран
	if (state == 1) {
		cout << "Компьютер: " << name << endl;
		cout << "Основные компоненты и их характеристики:" << endl;
		if (motherboard.check()) motherboard.print();
		else cout << "Материнская плата отсутствует" << endl;
		if (processor.check()) processor.print();
		else cout << "Процессор отсутствует" << endl;
		if (ram.check()) ram.print();
		else cout << "Оперативная память отсутствует" << endl;
		if (videocard.check()) videocard.print();
		else cout << "Видеокарта отсутствует" << endl;
		cout << endl;
	}
}

void Computer::set_name(string name) {  //ф-ция присвоения имени
	this->name = name;
}

void Computer::set_motherboard(Motherboard motherboard) {  //ф-ция присвоения материнской платы
	this->motherboard = motherboard;
}

void Computer::set_ram(Ram ram) {  //ф-ция присвоения оперативной памяти
	this->ram = ram;
}

void Computer::set_videocard(Videocard videocard) {  //ф-ция присвоения видеокарты
	this->videocard = videocard;
}

void Computer::set_processor(Processor processor) {  //ф-ция присвоения процессора
	this->processor = processor;
}

void emergency_turn_off(Computer comp) {
	comp.state = 0;
	cout << "Компьютер экстренно выключен";
}

int Computer::count = 0;

void Computer::print_count() {  //вывод кол-ва компьютеров
	cout << "Количество компьютеров: " << count << endl;
}

void Computer::set_count(int a) {  //установка кол-ва компьютеров
	count = a;
}

int Computer::get_count() { //получение кол-ва компьютеров
	return count;
}

File: JavaProgramm/src/com/company/Computer.java
package com.company;

public class Computer {

    static int count = 0; //кол-во компьютеров
    String name = "Без названия";  //название компьютера
    Processor processor = new Processor();  //процессор
    Ram ram = new Ram();  //оперативная память
    Videocard videocard = new Videocard();  //видеокарта
    Motherboard motherboard = new Motherboard();  //материнская плата
    int state = 0;  //состояние компбютера(0 - выключен)

    public Computer() {
        count = ++count;
    }

    public Computer(String name, Processor processor, Ram ram, Videocard videocard, Motherboard motherboard) {  //конструктор
        this.name = name;
        this.processor = processor;
        this.ram = ram;
        this.videocard = videocard;
        this.motherboard = motherboard;
        count = ++count;
    }

    public void turnon() {  //включить компьютер
        if (state == 0) {
            state = 1;
            System.out.println("Компьютер включен: " + name);
        }
        else System.out.println("Компьютер уже включен: " + name);
    }

    public void turnoff() {  //выключить компьютер
        if (state == 1) {
            state = 0;
            System.out.println("Компьютер выключен: " + name);
        }
    }

    public void check() {  //проверить и вывести состовляющие компьютера на экран
        if (state == 1) {
            System.out.println("Компьютер: " + name);
            System.out.println("Основные компоненты и их характеристики:");
            if (motherboard.check() == 1) motherboard.print();
            else System.out.println("Материнская плата отсутствует");
            if (processor.check() == 1) processor.print();
            else System.out.println("Процессор отсутствует");
            if (ram.check() == 1) ram.print();
            else System.out.println("Оперативная память отсутствует");
            if (videocard.check() == 1) videocard.print();
            else System.out.println("Видеокарта отсутствует");
            System.out.println();
        }
    }

    public void setName(String name) {  //метод присвоения имени
        this.name = name;
    }

    public void setMotherboard(Motherboard motherboard) {  //метод присвоения материнской платы
        this.motherboard = motherboard;
    }

    public void setRam(Ram ram) {  //метод присвоения оперативной памяти
        this.ram = ram;
    }

    public void setVideocard(Videocard videocard) {  //метод присвоения видеокарты
        this.videocard = videocard;
    }

    public void setProcessor(Processor processor) {  //метод присвоения процессора
        this.processor = processor;
    }

    public static void SetCount(int a)  //установка кол-ва компьютеров
    {
        count = a;
    }
    public static void PrintCount()  //вывод кол-ва компьютеров
    {
        System.out.println("Количество компьютеров: " + count);
    }
    public static int GetCount()  //получение кол-ва компьютеров
    {
        return count;
    }
}
